diff --git a/sites/all/modules/contrib/gmap/js/marker.js b/sites/all/modules/contrib/gmap/js/marker.js
index 2ce6e51..56fdd28 100644
--- a/sites/all/modules/contrib/gmap/js/marker.js
+++ b/sites/all/modules/contrib/gmap/js/marker.js
@@ -87,7 +87,7 @@ Drupal.gmap.addHandler('gmap', function (elem) {
             // If there was a callback, prefix that.
             // (If there wasn't, marker.rmt was the FULL path.)
             if (obj.vars.rmtcallback) {
-                uri = Drupal.settings.basePath + Drupal.settings.pathPrefix + obj.vars.rmtcallback + '/' + marker.rmt;
+                uri = obj.vars.rmtcallback + '/' + marker.rmt;
             }
             // @Bevan: I think it makes more sense to do it in this order.
             // @Bevan: I don't like your choice of variable btw, seems to me like
