diff --git a/sites/all/modules/contrib/multistep/multistep.module b/sites/all/modules/contrib/multistep/multistep.module
index cda6a0b..1cbc8d4 100644
--- a/sites/all/modules/contrib/multistep/multistep.module
+++ b/sites/all/modules/contrib/multistep/multistep.module
@@ -1082,6 +1082,7 @@ function _multistep_update_status($entity, $type, $status, $step = 'all') {
     }
   }
   elseif ($type == 'user' && !empty($entity->uid)) {
+    return;
     // When all steps are passed (or no step specified), initialize
     if ($step == 'all') {
       $groups = multistep_get_steps($entity, $type);
