diff --git a/sites/all/modules/contrib/multistep/multistep.module b/sites/all/modules/contrib/multistep/multistep.module
index ea4187a..e455c46 100644
--- a/sites/all/modules/contrib/multistep/multistep.module
+++ b/sites/all/modules/contrib/multistep/multistep.module
@@ -229,7 +229,7 @@ function multistep_block_view($type = '', $entity_type = 'node') {
         'id' => 'multistep-' . $entity_type . '-' . $type . '-' . $group->group_name,
         'class' => array(
           is_null($status) || $status == '' ? '' : 'multistep-' . $status,
-          $current_step == $group->group_name ? 'active': '',
+          $current_step == $group->group_name ? 'active-step': '',
         ),
         'url' => $id && ($workflow_mode == 'free' || ($workflow_mode == 'direct' && ($status == 'submitted' || multistep_get_next($type, $last_submitted_step) == $group->group_name))) ? $_GET['q'] : NULL,
         'step' => $group->group_name,
diff --git a/sites/all/modules/contrib/multistep/multistep.theme.inc b/sites/all/modules/contrib/multistep/multistep.theme.inc
index e87a993..2700d84 100644
--- a/sites/all/modules/contrib/multistep/multistep.theme.inc
+++ b/sites/all/modules/contrib/multistep/multistep.theme.inc
@@ -11,7 +11,7 @@ function theme_multistep_navigation($variables = array()) {
   foreach ($variables['links']['#children'] as $key => $group) {
     extract($group);
     if (isset($url) || $next_unsubmitted_step) {
-      $content[$key] = l($label, $url, array('attributes' => array('id' => $id, 'class' => $class), 'query' => array('step' => $step)));
+      $content[$key] = l($label, $url, array('html' => TRUE,'attributes' => array('id' => $id, 'class' => $class), 'query' => array('step' => $step)));
     }
     else {
       $content[$key] = '<span id="'. $id .'" class="' . implode(' ', $class) . '">' . $label . '</span>';
@@ -23,7 +23,7 @@ function theme_multistep_navigation($variables = array()) {
 /**
  * Block Theme function that displays the default output of a progress bar.
  * Use this theme function to override the output / display of this block.
- * 
+ *
  * for small prograss values <50% the percentage is shown to the right of the bar
  */
 function theme_multistep_progress_bar($variables = array()) {
