diff --git a/sites/all/modules/contrib/multistep/multistep.module b/sites/all/modules/contrib/multistep/multistep.module
index ea4187a..eeacb8b 100644
--- a/sites/all/modules/contrib/multistep/multistep.module
+++ b/sites/all/modules/contrib/multistep/multistep.module
@@ -674,7 +674,8 @@ function multistep_node_validate($node, $form, &$form_state) {
 function multistep_submit($form, &$form_state) {
 
   //Determine if it's a user or node
-  $node = isset($form_state['node']) && !empty($user = is_object($form_state['node']));
+  $user = isset($form_state['node']) && is_object($form_state['node']);
+  $node = isset($form_state['node']) && !empty($user);
   $user = isset($form_state['user']) && is_object($form_state['user']) && !empty($form_state['user']);
 
   if ($node) {
